src/hotspot/share/jfr/periodic/sampling/jfrThreadSampler.cpp

```diff
@@ -554,11 +554,11 @@ void JfrThreadSampler::run() {
       os::naked_sleep(sleep_to_next);
     }
 
-    if ((next_j - sleep_to_next) <= 0) {
+    if (next_j <= sleep_to_next) {
```

Comment:
It looks to me like this changes semantics so that it's now broken if `X_period_millis` wraps around. Do you actually want to change semantics in this case?

Reply:
Yes, if you mean the case where X_period_millis is max_jlong, next_X is near max_jlong, and the old subtract of a negative sleep_to_next would have caused an overflow, I don't think we want to produce a sample in that case. But I could be wrong.

I had comments explaining all the pathological cases, but I guess I removed them.  The other one is when next_j < 0, next_n < 0, sleep_to_next < 0, next_j != next_n.